### ISSUE TITLE
Accessing additional MODIS products using MODIStsp

### ISSUE THREAD

**Issue by bhick001** [ropensci/MODIStsp#196]
Is it possible to adapt this to acces additional MODIS products (e.g. atmospheric products like aerosol optical depth)?

**Comment by lbusett** [comment]
@bhick001 

sorry for the late answer. MODIStsp is designed to work with MODIS Land Products, as described here: https://lpdaac.usgs.gov/product_search/?collections=Combined+MODIS&collections=Terra+MODIS&collections=Aqua+MODIS&view=list

Are the data you are interested in within that list? 

Lorenzo

**Comment by bhick001** [comment]
Hello, 
Yes, it is "MCD19A2.006" | Combined MODIS | Aerosol Optical Depth | 1000 | Daily

**Comment by lbusett** [comment]
Then it should be possible to add it. I'll have a look as soon as I can.

**Comment by lbusett** [comment]
@bhick001 

I added the product in the RC for MODIStsp 1.4.0. 

The new version bring many updates since it deals with the switch to GDAL3/PROJ6. 
Would you be willing to test by installing from github using

```r
remotes::install_github("ropensci/MODIStsp", ref = "feature/gdal3")
```

?

**Comment by YunquanZhang** [comment]
I have installed versition 1.3.9, but I failed to install MODIStsp 1.4.0. using remotes::install_github("ropensci/MODIStsp", ref = "feature/gdal3")

**Comment by lbusett** [comment]
@YunquanZhang 

sorry for that. Could you report which kind of error you are getting while trying to install the github version? 

   Lorenzo

**Comment by YunquanZhang** [comment]
This package helps a lot. But I further want to download and deal with "MCD19A2.006" | Combined MODIS | Aerosol Optical Depth | 1000 | Daily product. 

I used "remotes::install_github("ropensci/MODIStsp", ref = "feature/gdal3")" to install versition 1.40. It reminds "Error: Failed to install 'MODIStsp' from GitHub:    schannel: failed to receive handshake, SSL/TLS connection failed".

**Comment by YunquanZhang** [comment]
This package helps a lot. But I further want to download and deal with "MCD19A2.006" | Combined MODIS | Aerosol Optical Depth | 1000 | Daily product.

I used "remotes::install_github("ropensci/MODIStsp", ref = "feature/gdal3")" to install versition 1.40. It reminds

 Downloading GitHub repo ropensci/MODIStsp@feature/gdal3
Skipping 16 packages not available: bitops, data.table, gdalUtilities, gWidgets, gWidgetsRGtk2, httr, jsonlite, raster, sf, stringr, xts, xml2, leaflet, shiny, mapview, mapedit
WARNING: Rtools is required to build R packages, but is not currently installed.

Please download and install Rtools custom from http://cran.r-project.org/bin/windows/Rtools/.
√  checking for file 'C:\Users\Administrator\AppData\Local\Temp\RtmpIhs5Xg\remotes1d6459864606\ropensci-MODIStsp-6cb7e76/DESCRIPTION' ...
-  preparing 'MODIStsp': (1.8s)
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts (906ms)
-  checking for empty or unneeded directories
-  building 'MODIStsp_1.3.9.9000.tar.gz'
   
ERROR: dependency 'gdalUtilities' is not available for package 'MODIStsp'
* removing 'D:/Program Files (x86)/R-3.6.3/library/MODIStsp'

**Comment by lbusett** [comment]
Hi @YunquanZhang 

  could you try installing Rtools, as suggested in the error message? 

You can find it here: 

https://cran.r-project.org/bin/windows/Rtools/history.html

(be sure to install the one corresponding to your "R" version)

Lorenzo

**Comment by YunquanZhang** [comment]
Thanks, I have succeed in installing the package after installing the Rtools. But I found that the installed  ‘MODIStsp’ version is 1.3.9.9000 not 1.40. Additionally,  another problem occurred when running "MODIStsp(gui = T)": GDAL version in use: 2.2.3   Error: $ operator is invalid for atomic vectors.  I can load the package by running "library(MODIStsp)". Thanks for your kind help in advance.

**Comment by lbusett** [comment]
Hi. The version number is correct. I'll have a look at the other problem. Thanks for reporting.

**Comment by bhick001** [comment]
I was unable to install the MODIStsp update, I can seem to install RGTK,

Warning: Failed to load RGtk2 dynamic library, attempting to install it.
Please install GTK+ from http://ftp.gnome.org/pub/gnome/binaries/win32/gtk+/2.22/gtk+-bundle_2.22.1-20101227_win32.zip
If the package still does not load, please ensure that GTK+ is installed and that it is on your PATH environment variable
IN ANY CASE, RESTART R BEFORE TRYING TO LOAD THE PACKAGE AGAIN
Error in inDL(x, as.logical(local), as.logical(now), ...) : 
  unable to load shared object 'C:/Users/BrandonHickman/Documents/R/win-library/3.6/RGtk2/libs/i386/RGtk2.dll':
  LoadLibrary failure:  The specified module could not be found.


However I have bene able to install and load RGtk2 into R. Apparently though GTK+ is not available for my version of R

> install.packages('GTK+')
Installing package into ‘C:/Users/BrandonHickman/Documents/R/win-library/3.6’
(as ‘lib’ is unspecified)
Warning in install.packages :
  package ‘GTK+’ is not available (for R version 3.6.1)

**Comment by lbusett** [comment]
Hi, 
 
 since this is a different problem, can I ask you to open a different issue? 
Before doing that, I'd suggest trying following instructions here: 

https://github.com/ropensci/MODIStsp#-installing-on-windows
